app/services/warm_cache.rb

```diff
@@ -0,0 +1,20 @@
+class WarmCache
+  def call
+    # Get all provider who have an api token that has been used in the past month
+    ProvidersForVendorAPICacheWarmingQuery.new.call.each do |provider|
+      # Find the api version they last used
+      api_version = VendorAPIRequest
+        .where(provider_id: provider.id)
+        .select("regexp_matches(request_path, '/api/v(.*)/applications') result")
+        .order(created_at: :desc)
+        .first&.result&.first
```

Comment:
Can we move this to a scope or class? I can envision something like "Oh we need to run one for a specific provider. Which version are they using? How to know?" then if is a scope would be easier to call in a console than copying and pasting the whole query.